### PR TITLE
Cookies

### PR DESCRIPTION
**Pull request by Nortamo**
Proper cookie banner, also removes GA cookies when consent is changed or if no consent has been granted. 

- Could still be customized more to have a better visual look, but this is ok. 
- Would also be good if some-one who has done more web-dev can tell me if there are some strange solutions here.
- https://csc-guide-preview.rahtiapp.fi/origin/coockies/
- See the API https://www.osano.com/cookieconsent/documentation/javascript-api/ for customization 
- Configuration wizard https://www.osano.com/cookieconsent/download/

- Embedded youtube videos are transformed into images with hyperlinks if consent has not been granted

### CODE REVIEW

**Comment by Nortamo** [comment]
`iframes` with youtube links also create some cookies which we should try to block (or disable embedded videos)

File: mkdocs_csc/disable_ga.html [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        if( consent == "allow"){
+            frames[idx].src=youtube_embed_link
+            frames[idx].removeAttribute("srcdoc")
+        }
+        else{
+        content =`
+            <style>
+            .top {
+                position: absolute;
+                left: 50%;
```
Comment (robinkar) [inline mkdocs_csc/disable_ga.html:52]:
```suggestion
                left: calc(50% - ${logo_w/2}px);
```
Could use calc to position the logo in the center

File: mkdocs_csc/disable_ga.html [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                top: 50%;
```
Comment (robinkar) [inline mkdocs_csc/disable_ga.html:53]:
```suggestion
                top: calc(50% - ${logo_h/2}px);
```
Use calc here too for centering, logo_h and logo_w have incorrect aspect ratio but it's close enough

File: mkdocs_csc/disable_ga.html [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                z-index: 1;
+            }
+            </style>
+
+            <a href=${youtube_link} target="_blank"><img src=https://img.youtube.com/vi/${youtube_id}/0.jpg width=${frames[idx].width} height=${frames[idx].height} ></a>
+            <a href=${youtube_link} target="_blank"><img class="top" width=${logo_w} height=${logo_h} src=https://www.pngkit.com/png/full/2-21145_youtube-logo-transparent-png-pictures-transparent-background-youtube.png></a>
```
Comment (robinkar) [inline mkdocs_csc/disable_ga.html:59]:
```suggestion
            <a href=${youtube_link} target="_blank"><img class="top" width=${logo_w} src=https://www.pngkit.com/png/full/2-21145_youtube-logo-transparent-png-pictures-transparent-background-youtube.png></a>
```
Omit the height attribute to retain the original aspect ratio of the logo

**Review by robinkar — COMMENTED** [review]
Here are some small changes to improve the styling of the Youtube placeholder

File: mkdocs_csc/disable_ga.html [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                    c = c.substring(1);
+                }
+                if (c.indexOf(name) == 0) {
+                    return c.substring(name.length, c.length);
+                }
+            }
+            return "";
+        }
+var consent=getCookie("cookieconsent_status")
+if(consent == "" || consent != "allow" ){
```
Comment (tdeneke) [inline mkdocs_csc/disable_ga.html:19]: Not that important but `consent == ""` looks redundant and can be removed.

File: mkdocs_csc/disable_ga.html [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        youtube_embed_link=`https://www.youtube.com/embed/${youtube_id}`
+        var consent=getCookie("cookieconsent_status")
+        logo_w=frames[idx].width /8;
+        logo_h=frames[idx].height/8;
+        
+        if( consent == "allow"){
+            frames[idx].src=youtube_embed_link
+            frames[idx].removeAttribute("srcdoc")
+        }
+        else{
```
Comment (tdeneke) [inline mkdocs_csc/disable_ga.html:47]: Just wondering if the ["Privacy Enhanced Mode"](https://support.google.com/youtube/answer/171780#zippy=%2Cturn-on-privacy-enhanced-mode) could be an option? More specifically, would some thing like `src="https://www.youtube-nocookie.com/embed/${youtube_id}"` be an alternative to `srcdoc`?

**Review by tdeneke — COMMENTED** [review]
Overall this looks good to me and my suggestions are optional.

File: mkdocs_csc/disable_ga.html (hunk 4 above)
Comment (Nortamo) [reply mkdocs_csc/disable_ga.html]: fixed

File: mkdocs_csc/disable_ga.html (hunk 5 above)
Comment (Nortamo) [reply mkdocs_csc/disable_ga.html:47]: Looks like that it still  dumps stuff somewhere else even if that is in use. So a total block seems to be the safest option.
https://axbom.com/embed-youtube-videos-without-cookies/

Yeah the srcdoc is kind of a hack, at some later point we could figure out some nicer way to conditionally load the iframes.

**Review by tdeneke — APPROVED** [review]

**Pull request merged**